rfd/0-job-worker.md

```diff
@@ -44,23 +78,152 @@ The worker library streams job output through an `io.ReadCloser`. The server wil
 
 #### gRPC API
 
-The gRPC API follows the worker library fairly closely. It does not require the user id in the messages because user id is derived from the client certificate used to make the connection. Additionally, the stream output is different since gRPC can't stream bytes, rather, it will stream one newline separated string at a time. Last, it has no need to be able to execute the job in a child process as that's only required by the library.
+The gRPC API follows the worker library fairly closely. It does not require the user id in the messages because user id is the client certificate's serial number, not explicitly provided. Additionally, the stream output format is slightly different since gRPC can't stream bytes, rather, it will stream one newline separated string at a time.
 
 ### Security Considerations
 
+#### TLS
+
+The following TLS configuration will be used, where `clientCAs` includes the ca cert that should sign all client certs and `crt` contains the server's cert and private key. Note that cipher suites are not used because they are [not configurable](https://go.dev/blog/tls-cipher-suites) in Go when using TLS 1.3. This configuration requires and validates client certificates for all connections.
+
 ```go
-cfg := &tls.Config{
+tls.Config{
     ClientAuth:   tls.RequireAndVerifyClientCert,
     ClientCAs:    clientCAs,
     Certificates: []tls.Certificate{crt},
     MinVersion:   tls.VersionTLS13,
 }
 ```
 
+#### Authorization
+
+The serial number of the client certificate is used as a unique user identifier and is implicitly required for all requests. It is used for authorization such that only the user that starts a job is permitted to stop it, get its status or output. This way the client doesn't have to explicitly provide any other kind of identification. The serial number is guaranteed to be unique for all certificates signed by a given authority thus making this a valid method to uniquely identify users.
+
+#### Non-Considerations
+
+At its core, this is a very dangerous service. It provides a facility for remote execution and, since it needs to be privileged to set up the namespaces, to do so as root. There are a number of things that could be done to make this safer, but are not requirements of the challenge, so they will not be implemented.
+
+Some examples of security improvements in a future version would be:
+
+- dropping privileges before executing a job
+- chroot into a separate, read-only filesystem
+- better handling of commands and arguments (e.g. proper escaping)
+
 ### CLI UX
 
-### Implementation Details
+A single binary, `job-worker` is used for all actions. It has several subcommands for each role. All configuration is done via cli flags.
+
+#### root level usage output
+
+```
+A prototype job worker service that provides an api to run arbitrary linux processes
+
+Usage:
+  job-worker [command]
+
+Available Commands:
+  completion  Generate the autocompletion script for the specified shell
```

Comment:
Feel free to drop this from the implementation.

Reply:
it's just part of the cobra library, i'm not planning on doing anything extra for its functionality